include/mxnet/kvstore.h

```diff
@@ -155,6 +156,29 @@ class KVStore {
                     const std::vector<NDArray*>& values,
                     int priority = 0) = 0;
 
+  /*!
+   * \brief pull a list of key-value pairs from the store.
+   *        The NDArray pulled back will be in row_sparse storage with only the
+   *        specified row_ids present (others rows are zeros).
+   * \param keys the list of keys
+   * \param values the list of buffers - row_id pairs
+   * \param priority the priority of the action.
+   */
+  virtual void PullRowSparse(const std::vector<int>& str_keys,
+                             const std::vector<std::pair<NDArray*, NDArray>>& val_rowids,
```

Comment:
For each value we may have different row_ids to pull. e.g.
val0.ctx = mx.gpu(0), val1.ctx = mx.gpu(1)
kv.row_sparse_pull("key", [val0, val1], [mx.nd.array([0, 1, 3]), mx.nd.array([4,5,6])]